package/MDAnalysis/core/topology.py

```diff
@@ -122,8 +122,12 @@ def r2a_1d(self, rix):
             indices of atoms present in residues, collectively
 
         """
-        return np.concatenate([self._atom_order[x:y]
-                               for x, y in self._res_ptrs[rix]])
+        if isinstance(rix, int):
```

Comment:
Can this be wrapped in a try/except block instead? Iirc, there's less overheads for these, so if Group can go through the `try` half, it shouldn't affect performance, and Component can be the `except` half